src/Common/CurrentThread.h

```diff
@@ -102,6 +110,39 @@ class CurrentThread
         void logPeakMemoryUsage();
         bool log_peak_memory_usage_in_destructor = true;
     };
+
+    /// Scoped attach/detach of IO resource links
+    struct IOScope : private boost::noncopyable
```

Comment:
I think we will use this scope eventually for all IO operations including local, hence the name.

Reply:
Do we haves cases with only one type of resources attached? Or they are used together always?

Reply:
For local IO it could be only read or write.